diff --git a/source/public/kernel/nvidia/include/media/camera_common.h b/source/public/kernel/nvidia/include/media/camera_common.h
old mode 100644
new mode 100755
index 22a41bc..882754d
--- a/source/public/kernel/nvidia/include/media/camera_common.h
+++ b/source/public/kernel/nvidia/include/media/camera_common.h
@@ -238,6 +238,7 @@ struct camera_common_data {
 	/* TODO: cleanup neeeded once all the sensors adapt new framework */
 	struct tegracam_ctrl_handler		*tegracam_ctrl_hdl;
 	struct regmap				*regmap;
+	struct i2c_client *i2c_client;
 	struct camera_common_pdata		*pdata;
 	/* TODO: cleanup needed for priv once all the sensors adapt new framework */
 	void	*priv;
