diff --git a/source/kernel/kernel-jammy-src/drivers/i2c/busses/i2c-tegra.c b/source/kernel/kernel-jammy-src/drivers/i2c/busses/i2c-tegra.c
index cd50436..e420425 100644
--- a/source/kernel/kernel-jammy-src/drivers/i2c/busses/i2c-tegra.c
+++ b/source/kernel/kernel-jammy-src/drivers/i2c/busses/i2c-tegra.c
@@ -142,6 +142,8 @@
 /* packet header size in bytes */
 #define I2C_PACKET_HEADER_SIZE			12
 
+#define I2C_ADAPTER_TIMEOUT_INIT (6 * HZ) // DAL custom code
+
 /*
  * I2C Controller will use PIO mode for transfers up to 32 bytes in order to
  * avoid DMA overhead, otherwise external APB DMA controller will be used.
@@ -1372,6 +1374,21 @@ static int tegra_i2c_xfer_msg(struct tegra_i2c_dev *i2c_dev,
 	 * Transfer time in mSec = Total bits / transfer rate
 	 * Total bits = 9 bits per byte (including ACK bit) + Start & stop bits
 	 */
+	
+	 // DAL custom code start
+	if (i2c_dev->adapter.timeout != I2C_ADAPTER_TIMEOUT_INIT) // If value changed from init value, modify the the transfer timeout
+	{
+		if (i2c_dev->adapter.timeout < 0)
+		{
+			xfer_time = UINT_MAX;
+		}
+		else if (i2c_dev->adapter.timeout > 0)
+		{
+			xfer_time = i2c_dev->adapter.timeout;
+		}
+		// Value 0 = keep defaut transfer timeout
+	}
+	 // DAL custom code end
 	xfer_time += DIV_ROUND_CLOSEST(((xfer_size * 9) + 2) * MSEC_PER_SEC,
 				       i2c_dev->bus_clk_rate);
 
@@ -1951,7 +1968,7 @@ static int tegra_i2c_probe(struct platform_device *pdev)
 	i2c_dev->adapter.dev.of_node = i2c_dev->dev->of_node;
 	i2c_dev->adapter.dev.parent = i2c_dev->dev;
 	i2c_dev->adapter.retries = 1;
-	i2c_dev->adapter.timeout = 6 * HZ;
+	i2c_dev->adapter.timeout = I2C_ADAPTER_TIMEOUT_INIT; // DAL custom code
 	i2c_dev->adapter.quirks = i2c_dev->hw->quirks;
 	i2c_dev->adapter.owner = THIS_MODULE;
 	i2c_dev->adapter.class = I2C_CLASS_DEPRECATED;
diff --git a/source/kernel/kernel-jammy-src/drivers/i2c/i2c-mux.c b/source/kernel/kernel-jammy-src/drivers/i2c/i2c-mux.c
index c90cec8..5255ce9 100644
--- a/source/kernel/kernel-jammy-src/drivers/i2c/i2c-mux.c
+++ b/source/kernel/kernel-jammy-src/drivers/i2c/i2c-mux.c
@@ -48,7 +48,10 @@ static int __i2c_mux_master_xfer(struct i2c_adapter *adap,
 
 	ret = muxc->select(muxc, priv->chan_id);
 	if (ret >= 0)
+	{
+		parent->timeout = adap->timeout; // DAL custom code
 		ret = __i2c_transfer(parent, msgs, num);
+	}
 	if (muxc->deselect)
 		muxc->deselect(muxc, priv->chan_id);
 
